### DISCUSSION TITLE
[Discussion] Concept clarification

### DISCUSSION THREAD

**Discussion by TanukiSharp** [dotnet/maui#42] [General]
I'm truly in love with the concept of multi-platform UI framework, but the README says (at the time I write) "Evolution of Xamarin.Forms". Can this be clarified please ?

On one hand, it sounds like you are turning XF into a multi-platform UI framework, on the other hand, looking at the file structure, there seems to be stuffs related to WPF, UWP and so on. So the intent is not clear.

Maybe just the README is misleading and the part "Evolution of Xamarin.Forms" should be elaborated as "Evolution of Xamarin.Forms for the mobile platforms part" ?

**Comment by davidortinau** [comment]
Hi @TanukiSharp, I'll try to clarify. I can see how the readme may expect some history of Xamarin.Forms knowledge.

Xamarin.Forms has long focused heavily on mobile and the initial support for Windows mobile grew into having a UWP that could also target desktop. Still, the focus was on mobile. 

macOS, WPF, and Linux were all added via experiments or community contributions. As such, this transition to .NET Multi-platform App UI is a more clear declaration to not only support mobile but desktop as well starting with macOS and Windows for .NET 6.

The current file structure is Xamarin.Forms brought over, and the first step has been to refactor the namespaces. Many changes are planned for revising the structure, removing duplication, taking advantage of multi-targeting, and generally cleaning up. We will have a future spec that outlines that plan. So the structure today does not at all express the intent, you're right.

Although WPF for example exists in the code, the maintenance of that platform, a separate project from Core, is mostly isolated from the rest of the product code and is open for the community to support.

The [platform status page](https://github.com/dotnet/maui/wiki/Status) also shows the lack of parity among the platforms available, and tracks our progress to closing that gap, especially for desktop.

**Comment by TanukiSharp** [comment]
Thanks for your answer.

So if I understand correctly, it's really extending XF to make it works on desktops too, right ?

Sorry if I'm wrong.

**Comment by JohnH-FF** [comment]
XF already works on MacOS and UWP (I have written a prototype of our primary app that works on all 4 platforms), but I am hoping MAUI will make them first class citizens like iOS and Android.

**Comment by TanukiSharp** [comment]
(I'm going further in the conversation before @davidortinau answers, sorry)

What I meant with my question is, why not extending WPF to the mobile world instead of extending XF to the desktop world ?

Not like I want to sound mean, but from what I could experience by myself, WPF is much more well designed than XF, so it would be a better choice in my opinion. Again, this is just my opinion, and I'm not saying WPF has no design issues.

**Comment by scherenhaenden** [comment]
> Hi @TanukiSharp, I'll try to clarify. I can see how the readme may expect some history of Xamarin.Forms knowledge.
> 
> Xamarin.Forms has long focused heavily on mobile and the initial support for Windows mobile grew into having a UWP that could also target desktop. Still, the focus was on mobile.
> 
> macOS, WPF, and Linux were all added via experiments or community contributions. As such, this transition to .NET Multi-platform App UI is a more clear declaration to not only support mobile but desktop as well starting with macOS and Windows for .NET 6.
> 
> The current file structure is Xamarin.Forms brought over, and the first step has been to refactor the namespaces. Many changes are planned for revising the structure, removing duplication, taking advantage of multi-targeting, and generally cleaning up. We will have a future spec that outlines that plan. So the structure today does not at all express the intent, you're right.
> 
> Although WPF for example exists in the code, the maintenance of that platform, a separate project from Core, is mostly isolated from the rest of the product code and is open for the community to support.
> 
> The [platform status page](https://github.com/dotnet/maui/wiki/Status) also shows the lack of parity among the platforms available, and tracks our progress to closing that gap, especially for desktop.


"Linux were all added via experiments or community contributions." that is the one experiment taken from MS that is now NOT available for us, the users of the Nix* world...
